ballsdex/packages/trade/cog.py

```diff
@@ -427,16 +428,43 @@ async def history(
     async def view(
         self,
         interaction: discord.Interaction["BallsDexBot"],
+        trade_id: str | None = None,
     ):
         """
         View the countryballs added to an ongoing trade.
+
+        Parameters
+        ----------
+        trade_id: str | None
+            The id of the trade to view, current if empty.
         """
-        trade, trader = self.get_trade(interaction)
-        if not trade or not trader:
-            await interaction.response.send_message(
-                "You do not have an ongoing trade.", ephemeral=True
-            )
-            return
+        if trade_id:
+            try:
+                trade = await TradeModel.get(id=trade_id).prefetch_related("player1", "player2")
```

Comment:
You forgot to convert from base 16
```suggestion
                trade = await TradeModel.get(id=int(trade_id, 16)).prefetch_related("player1", "player2")
```